File: python/26_Remove_Duplicates_from_Sorted_Array.py
class Solution:
    # @param {integer[]} nums
    # @return {integer}
    def removeDuplicates(self, nums):
        if not nums:
            return 0
        last=nums[0]
        m=1
        for i in range(1, len(nums)):
            if nums[i]!=last:
                last=nums[i]
                nums[m]=nums[i]
                m+=1
        del nums[m:]
        return len(nums)

File: Java/26_Remove_Duplicates_from_Sorted_Array.java
public class Solution {
    public int removeDuplicates(int[] nums) {
        if(nums==null)return 0;
        int len=nums.length;
        int cur=0,runner=0;
        while(runner<len){
            nums[cur]=nums[runner];// 存储
            while(runner<len-1&&nums[runner]==nums[runner+1]){
                runner++;// 跳过重复
            }
            cur++;// 前移
            runner++;// 前移
        }
        return cur;
    }
}